src/me/mrCookieSlime/Slimefun/Variables.java

```diff
@@ -18,6 +21,7 @@
 	
 	public static Map<UUID, Integer> enchanting = new HashMap<UUID, Integer>();
 	public static Map<UUID, ItemStack> backpack = new HashMap<UUID, ItemStack>();
+	public static Map<Location, Boolean> altarinuse = new HashMap<Location, Boolean>();
```

Comment:
I can switch it to a Hashset easily enough.  But before I do I want to try to figure out if I can even make the second proposed solution work since I started in that direction but then hit a wall rather quickly.  If I'm understanding you - as this is what I first thought: HashMap<ALTARLocation, PEDESTALLocation[]>, yes?  But on the first click, nothing has been populated yet, so I still have to do location calculations to put the key for the pedestal that was just clicked then assign that pedestal to that array.  The second and third pedestal... basically, I still have to do the same thing since I've no way of knowing of those pedestals belong to that altar until I check to see which alter they should have.  I could do a second check with the previously collected data... and only do a calculation search if a match was not found but... I think this just adds a step.  I thought maybe to simplify the search/result process that instead it could be a HashMap<BlockLocation, Boolean IsAnAltar, AltarLocation>  whereas if it is not an altar, it would simply say where the altar was... thereby skipping any unnecessary reuse of the location calculations.  This only solves effienciency for spam reclicking though.  Which might be an alright thing to do here.  Would still want to remove all the data after use in case the player rebuilds the thing in such a way to work around the patch.  In any case, any further thoughts or pointers before I give myself a headache with this would be appreciated, thanks!

Reply:
You cannot have a HashMap with 3 generic types, HashMaps represent Key/Value Pairs.

And upon first right click, all pedestals are available right here:
https://github.com/TheBusyBiscuit/Slimefun4/blob/master/src/me/mrCookieSlime/Slimefun/listeners/AncientAltarListener.java#L86

Reply:
going to retest i may have this working without the altar search if so i'll commit the adjusteed files.

Reply:
Yeah that does it still... I was concerned that the spam clicking thing was just getting around the pedestal checks and just starting over each time... this still bogs down the animation if I spam click a pedestal in spite of making the computer actually work a lot less at checking and locating before stopping the actions, not really sure why.  Stopped checking for preexistence of the values since HashSets don't care.  Anyway see if this adjustment works for you.